src/Leaflet/Core/Layer.purs

```diff
@@ -0,0 +1,275 @@
+  ∷ ∀ e r. Fn3 (Array (Array T.Degrees)) ConvertDict r (Eff (dom ∷ DOM|e) T.Polyline)
+
+foreign import polygon_
+  ∷ ∀ e r. Fn3 (Array (Array T.Degrees)) ConvertDict r (Eff (dom ∷ DOM|e) T.Polygon)
+
+foreign import rectangle_
+  ∷ ∀ e r. Fn3 (Array (Array T.Degrees)) ConvertDict r (Eff (dom ∷ DOM|e) T.Rectangle)
+
+foreign import on_
+  ∷ ∀ e. Fn3 String (T.Event → Eff (dom ∷ DOM|e) Unit) T.Evented (Eff (dom ∷ DOM|e) Unit)
+
+foreign import setIcon_
+  ∷ ∀ e. Fn2 T.Icon T.Marker (Eff (dom ∷ DOM|e) T.Marker)
+
+foreign import addLayer_
+  ∷ ∀ e. Fn2 T.Layer T.Leaflet (Eff (dom ∷ DOM|e) T.Leaflet)
+
+foreign import removeLayer_
+  ∷ ∀ e. Fn2 T.Layer T.Leaflet (Eff (dom ∷ DOM|e) T.Leaflet)
+
+
+layer
+  ∷ ∀ e m
+  . MonadEff (dom ∷ DOM|e) m
+  ⇒ m T.Layer
+layer = liftEff layer_
+
+tileLayer
+  ∷ ∀ e m
+  . MonadEff (dom ∷ DOM|e) m
+  ⇒ URIRef
+  → m T.TileLayer
+tileLayer =
+  liftEff ∘ tileLayer_ ∘ converter.printURI
+
+marker
+  ∷ ∀ e m
+  . MonadEff (dom ∷ DOM|e) m
+  ⇒ T.LatLng
+  → m T.Marker
+marker latLng =
+  liftEff $ marker_ $ converter.convertLatLng latLng
+
+icon
+  ∷ ∀ e r1 r2 m
+  . Union r1 r2 T.IconConf
+  ⇒ MonadEff (dom ∷ DOM|e) m
+  ⇒ Record r1
+  → m T.Icon
+icon r =
+  liftEff $ runFn2 icon_ converter r
+
+popup
+  ∷ ∀ e r1 r2 m
+  . MonadEff (dom ∷ DOM|e) m
+  ⇒ Union r1 r2 T.PopupConf
+  ⇒ Record r1
+  → m T.Popup
+popup r =
+  liftEff $ runFn2 popup_ converter r
+
+setLatLng
+  ∷ ∀ e m
+  . MonadEff (dom ∷ DOM|e) m
+  ⇒ T.LatLng
+  → T.Popup
+  → m T.Popup
+setLatLng latLng p =
+  liftEff $ runFn2 setLatLng_ (converter.convertLatLng latLng) p
+
+setContent
+  ∷ ∀ e m
+  . MonadEff (dom ∷ DOM|e) m
+  ⇒ String
+  → T.Popup
+  → m T.Popup
+setContent c p =
+  liftEff $ runFn2 setContent_ c p
+
+openOn
+  ∷ ∀ e m
+  . MonadEff (dom ∷ DOM|e) m
+  ⇒ T.Leaflet
+  → T.Popup
+  → m T.Popup
+openOn m p =
+  liftEff $ runFn2 openOn_ m p
+
+circleMarker
+  ∷ ∀ e r1 r2 m
+  . MonadEff (dom ∷ DOM|e) m
+  ⇒ Union r1 r2 T.CircleConf
+  ⇒ T.LatLng
+  → Record r1
+  → m T.CircleMarker
+circleMarker ll r =
+  liftEff $ runFn3 circleMarker_ (converter.convertLatLng ll) converter r
+
+circle
+  ∷ ∀ e r1 r2 m
+  . MonadEff (dom ∷ DOM|e) m
+  ⇒ Union r1 r2 T.CircleConf
+  ⇒ T.LatLng
+  → Record r1
+  → m T.Circle
+circle ll r =
+  liftEff $ runFn3 circle_ (converter.convertLatLng ll) converter r
+
+polyline
+  ∷ ∀ e r1 r2 m f
+  . MonadEff (dom ∷ DOM|e) m
+  ⇒ Union r1 r2 (T.PolylineConf ())
+  ⇒ Foldable f
+  ⇒ f T.LatLng
+  → Record r1
+  → m T.Polyline
+polyline lls r =
+  liftEff $ runFn3 polyline_ (foldMap (A.singleton ∘ converter.convertLatLng) lls) converter r
+
+polygon
+  ∷ ∀ e r1 r2 m f
+  . MonadEff (dom ∷ DOM|e) m
+  ⇒ Union r1 r2 (T.PolylineConf ())
+  ⇒ Foldable f
+  ⇒ f T.LatLng
+  → Record r1
+  → m T.Polygon
+polygon lls r =
+  liftEff $ runFn3 polygon_ (foldMap (A.singleton ∘ converter.convertLatLng) lls) converter r
+
+rectangle
+  ∷ ∀ e r1 r2 m
+  . MonadEff (dom ∷ DOM|e) m
+  ⇒ Union r1 r2 (T.PolylineConf ())

```

Comment:
So on stuff like this, there are no _required_ options for the confs, right? For any of them `{}` is a valid configuration?